### PR TITLE
Adds negative validation loss as a metric

### PR DESCRIPTION
**Pull request by jdunnmon**
- [x] Compute validation loss during training loop
- [x] Use negative validation loss as a tuning metric
- [x] Fixed running loss bug (adding multiplication by batch length)
- [x] Adding `_get_loss_fn` stub to `Classifier`
- [x] Modifying `LogWriter` to create a new log subdirectory for events and logs from every individual run, but still save the tuner report and best model to the root log directory

@bhancock8 a couple of thoughts:

(1) We could integrate the validation loss by conditionally switching the direction of the comparator in the tuner.  I thought it was easier for now to just make negative validation loss the parameter we maximize. I am open to adjusting that for the merge.

(2) I will fix the `.isort.cfg` file before merge -- for some reason it always updates to this locally for me.

### CODE REVIEW

File: metal/tuners/tuner.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def _clear_state(self):
         self.start_time = time()
         self.run_stats = []
         self.best_index = -1
-        self.best_score = -1
+        self.best_score = -1e9
```
Comment (bhancock8) [inline metal/tuners/tuner.py]: What's the motivation for this change?

File: metal/classifier.py [hunk 2]
```diff
@@ -195,6 +195,9 @@ def _create_checkpointer(self, checkpoint_config):
             model_class, **checkpoint_config, verbose=self.config["verbose"]
         )
 
+    def get_loss_fn(self):
```
Comment (bhancock8) [inline metal/classifier.py]: This needs an underscore. Also add a one-liner docstring describing the purpose of this and an error message with the exception that says it's unimplemented because it's an abstract class, not because we just haven't written it yet.

File: metal/classifier.py [hunk 3]
```diff
@@ -304,7 +307,7 @@ def _train_model(
                 optimizer.step()
 
                 # Keep running sum of losses
-                epoch_loss += loss.item()
+                epoch_loss += loss.item() * len(data[0])
```
Comment (bhancock8) [inline metal/classifier.py]: This feels obtuse; we average the loss, then multiply it back out, then average it again two turns later. I recognize that this change is just fixing a problem that was already there, but I think we should just work with the raw (sum) loss everywhere and then average once (for reporting/logging). Can you make a quick issue to remind us of this inconsistency?

File: metal/classifier.py [hunk 4]
```diff
@@ -322,6 +325,10 @@ def _train_model(
                     "validation_scoring_kwargs"
                 ]
                 self.eval()
+
+                # Adding loss function to support validating on loss
```
Comment (bhancock8) [inline metal/classifier.py]: What is this comment referring to/why is it hanging?

File: metal/classifier.py (hunk 3 above)
Comment (bhancock8) [reply metal/classifier.py]: Actually, I'll go ahead and add the issue.

File: metal/tuners/tuner.py (hunk 1 above)
Comment (jdunnmon) [reply metal/tuners/tuner.py]: Negative loss can be < -1, so makes sure we update.

File: metal/classifier.py (hunk 2 above)
Comment (jdunnmon) [reply metal/classifier.py]: Yep, will fix this.

File: metal/classifier.py (hunk 3 above)
Comment (jdunnmon) [reply metal/classifier.py]: Agree.

File: metal/classifier.py (hunk 4 above)
Comment (jdunnmon) [reply metal/classifier.py]: Silliness. I'll get rid of it :).

**Comment by jdunnmon** [comment]
Yep, I think that makes sense.  I'll get the tensorboard writer fix and the _get_loss_fn stub into master, o/w will move neg_loss changes into slicing branch.
